In [1]:
using LinearAlgebra
using SymPy

\[1\]

In [2]:
X = Rational[
    4321 4322 4323
    4324 4325 432_7
    4327 4328 4329
]

Int(det(X))

6

\[2\]

In [3]:
@vars a b c

Y = [
    4321 4322 4323+a
    4324 4325 4326+b
    4327 4328 4329+c
]

det(Y)

-3*a + 6*b - 3*c

\[3\]

In [4]:
A = Rational[
     3 1 3
    -1 3 2
    -1 2 3
]

Sym(det(A)) |> display
Sym.(det(A)*inv(A')) |> display
Sym.(inv(A))

19

3×3 Array{Sym,2}:
  5   1   1
  3  12  -7
 -7  -9  10

3×3 Array{Sym,2}:
 5/19   3/19  -7/19
 1/19  12/19  -9/19
 1/19  -7/19  10/19

\[4\]

In [5]:
A = Sym[
    3 -1 2
    4  1 3
    3  3 5
]

b = Sym[
    1
    1
    -1
]

A |> display
b |> display
A\b

3×3 Array{Sym,2}:
 3  -1  2
 4   1  3
 3   3  5

3-element Array{Sym,1}:
  1
  1
 -1

3-element Array{Sym,1}:
  12/17
  -1/17
 -10/17

In [6]:
det(A) |> display
A1 = copy(A); A1[:,1] = b
A2 = copy(A); A2[:,2] = b
A3 = copy(A); A3[:,3] = b
A1 |> display
A2 |> display
A3 |> display
[
    det(A1)/det(A)
    det(A2)/det(A)
    det(A3)/det(A)
]

17

3×3 Array{Sym,2}:
  1  -1  2
  1   1  3
 -1   3  5

3×3 Array{Sym,2}:
 3   1  2
 4   1  3
 3  -1  5

3×3 Array{Sym,2}:
 3  -1   1
 4   1   1
 3   3  -1

3-element Array{Sym,1}:
  12/17
  -1/17
 -10/17

\[5\]

In [7]:
a = Sym[
    4
    2
    -2
    1
]

b = Sym[
    2
    -4
    2
    1
]

c = dot(a,b)/norm(a)/norm(b)

-3/25

In [8]:
acos(Float64(c))/π*180

96.89210257934639

96.9度は直角に近い.

\[6\]

一般に $m\times n$ 行列 $X$ と $n\times m$ 行列 $Y$ について $(XY)^T=Y^T X^T$ が成立している. ゆえに, $AA^{-1}=A^{-1}A=E$ 全体の転置を取ると, $(A^{-1})^TA^T=A^T(A^{-1})^T=E$. これで $(A^{-1})^T$ が $A^T$ の逆行列になっていることが示された.

\[7\]

$A^T=A$ より $(A^{-1})^T = (A^T)^{-1}=A^{-1}$ となる. 一般に $b^T x = x^T b$ となる. ゆえに,

$$
\begin{aligned}
\mathrm{RHS} &=
(x^T + b^T A^{-1})(Ax+b) - b^T A^{-1}b
\\ &=
x^T A x + x^T b + b^T x + b^T A^{-1}b - b^T A^{-1}b
\\ &=
x^t A x + 2b^T x 
\\ &=
\mathrm{LHS}.
\end{aligned}
$$

\[8\]

In [9]:
K = Rational[
    5 2 1 3 2
    3 8 8 4 3
    6 4 2 4 2
    1 8 8 3 2
    5 8 8 5 4
]

rank(K)

3

\[9\]

In [10]:
L = [
    -1 6
    -2 6
]

Sym.(L)

2×2 Array{Sym,2}:
 -1  6
 -2  6

In [11]:
(L/3)^100

2×2 Array{Float64,2}:
 -3.0  6.0
 -2.0  4.0

In [12]:
P = [
    3 2
    2 1
]

D = [
    3 0
    0 2
]

P*D*Sym.(P)^(-1)

2×2 Array{Sym,2}:
 -1  6
 -2  6

In [13]:
@vars n integer=true
Dn = [
    3^n 0
    0 2^n
]

P*Dn*Sym.(P)^(-1)

2×2 Array{Sym,2}:
 4*2^n - 3*3^n  -6*2^n + 6*3^n
 2*2^n - 2*3^n  -3*2^n + 4*3^n

これに $3^{-n}$ をかけて $n\to\infty$ とすると,

$$
\begin{bmatrix}
-3 & 6 \\
-2 & 4 \\
\end{bmatrix}
$$

に収束する.